Expensify/App
# Fix signin page layout

### Details
The way the `SigninPageLayout` component was originally written was confusing and required bits and pieces of the forms that were its children to be located across separate components. 

Since we eventually want to refactor the `LoginForm` to be a single component (without wide/narrow counterparts), I've refactored the `SigninPageLayout` to be easier to use with single component forms as children.

### Fixed Issues
Expensify/Expensify#157816

### QA Testing Steps
1. Navigate to the sign in page.
2. Verify that it looks the same as they do on production.
3. Enter an email for an account that's already been created.
4. Verify that the password entry page looks the same as it does on production.
3. Create a new account with an email you have access to (I recommend using [EmailOnDeck](www.emailondeck.com)).
4. Verify that resend validation link screen looks the same as it does on production.
5. For Web, Mobile Web, iOS, and Android, use the link in the email that was sent to access the set password page.
6. Verify that the set password page looks the same as it does on production.

### Tested On


### Screenshots


#### Web
<img width="1792" alt="Screen Shot 2021-04-08 at 11 18 34 AM" src="https://user-images.githubusercontent.com/31285285/113963716-b7067900-985c-11eb-8d5b-14482da3fd0b.png">
<img width="1792" alt="Screen Shot 2021-04-08 at 11 18 44 AM" src="https://user-images.githubusercontent.com/31285285/113963725-bb329680-985c-11eb-8742-2c3d30a35ba7.png">
<img width="1792" alt="Screen Shot 2021-04-08 at 11 18 58 AM" src="https://user-images.githubusercontent.com/31285285/113963730-bc63c380-985c-11eb-8f06-680ac0179232.png">
<img width="1792" alt="Screen Shot 2021-04-08 at 11 22 03 AM" src="https://user-images.githubusercontent.com/31285285/113963732-bcfc5a00-985c-11eb-9707-a1755f0aee7b.png">

#### Mobile Web
<img width="507" alt="Screen Shot 2021-04-08 at 11 19 58 AM" src="https://user-images.githubusercontent.com/31285285/113963759-c7b6ef00-985c-11eb-92f1-9c3837510dd5.png">
<img width="479" alt="Screen Shot 2021-04-08 at 11 19 39 AM" src="https://user-images.githubusercontent.com/31285285/113963766-cc7ba300-985c-11eb-9785-32a60ea904b1.png">
<img width="486" alt="Screen Shot 2021-04-08 at 11 19 33 AM" src="https://user-images.githubusercontent.com/31285285/113963768-cdacd000-985c-11eb-8c8c-fde788ba4ae4.png">
<img width="496" alt="Screen Shot 2021-04-08 at 11 19 24 AM" src="https://user-images.githubusercontent.com/31285285/113963771-ce456680-985c-11eb-8a0d-e9769da35f98.png">

#### Desktop (set password page can't be accessed via deep links on desktop)
<img width="1194" alt="Screen Shot 2021-04-08 at 11 12 11 AM" src="https://user-images.githubusercontent.com/31285285/113963802-d8fffb80-985c-11eb-882b-6fed20cd2379.png">
<img width="1196" alt="Screen Shot 2021-04-08 at 11 12 07 AM" src="https://user-images.githubusercontent.com/31285285/113963794-d56c7480-985c-11eb-9f72-d5c0a6f37b46.png">
<img width="1200" alt="Screen Shot 2021-04-08 at 11 24 12 AM" src="https://user-images.githubusercontent.com/31285285/113963862-fb921480-985c-11eb-8cc5-617467ce4a31.png">

#### iOS
<img width="509" alt="Screen Shot 2021-04-08 at 11 06 57 AM" src="https://user-images.githubusercontent.com/31285285/113963895-09479a00-985d-11eb-92e4-188906ba2b99.png">
<img width="515" alt="Screen Shot 2021-04-08 at 11 06 53 AM" src="https://user-images.githubusercontent.com/31285285/113963909-0e0c4e00-985d-11eb-9b48-847d5d33b8f7.png">
<img width="506" alt="Screen Shot 2021-04-08 at 11 07 11 AM" src="https://user-images.githubusercontent.com/31285285/113963912-0f3d7b00-985d-11eb-8768-9d60f23caae1.png">
<img width="509" alt="Screen Shot 2021-04-08 at 11 07 28 AM" src="https://user-images.githubusercontent.com/31285285/113963913-0fd61180-985d-11eb-9d5e-843698b16f16.png">

#### Android
<img width="409" alt="Screen Shot 2021-04-08 at 11 17 52 AM" src="https://user-images.githubusercontent.com/31285285/113963951-211f1e00-985d-11eb-9645-841df3fdb6ba.png">
<img width="412" alt="Screen Shot 2021-04-08 at 11 18 27 AM" src="https://user-images.githubusercontent.com/31285285/113963960-241a0e80-985d-11eb-80aa-6480ae24f01b.png">
<img width="394" alt="Screen Shot 2021-04-08 at 11 18 16 AM" src="https://user-images.githubusercontent.com/31285285/113963967-25e3d200-985d-11eb-8417-e1253f82adc0.png">
<img width="431" alt="Screen Shot 2021-04-08 at 11 20 34 AM" src="https://user-images.githubusercontent.com/31285285/113963969-267c6880-985d-11eb-9b2b-1cd480e34924.png">
